core/deployment/src/main/java/io/quarkus/deployment/dev/testing/JunitTestRunner.java

```diff
@@ -0,0 +1,652 @@
+                        Throwable throwable = testExecutionResult.getThrowable().get();
+                        if (testClass != null) {
+                            //first we cut all the platform stuff out of the stack trace
+                            StackTraceElement[] st = throwable.getStackTrace();
+                            for (int i = st.length - 1; i >= 0; --i) {
+                                StackTraceElement elem = st[i];
+                                if (elem.getClassName().equals(testClass.getName())) {
+                                    StackTraceElement[] newst = new StackTraceElement[i + 1];
+                                    System.arraycopy(st, 0, newst, 0, i + 1);
+                                    st = newst;
+                                    break;
+                                }
+                            }
+
+                            //now cut out all the restassured internals
+                            //TODO: this should be pluggable
+                            for (int i = st.length - 1; i >= 0; --i) {
+                                StackTraceElement elem = st[i];
+                                if (elem.getClassName().startsWith("io.restassured")) {
+                                    StackTraceElement[] newst = new StackTraceElement[st.length - i];
+                                    System.arraycopy(st, i, newst, 0, st.length - i);
+                                    st = newst;
+                                    break;
+                                }
+                            }
+                            throwable.setStackTrace(st);
+                        }
+                    }
+                }
+
+                @Override
+                public void reportingEntryPublished(TestIdentifier testIdentifier, ReportEntry entry) {
+
+                }
+            });
+            if (aborted) {
+                return;
+            }
+            testState.updateResults(resultsByClass);
+            if (classScanResult != null) {
+                testState.classesRemoved(classScanResult.getDeletedClassNames());
+            }
+
+            QuarkusConsole.INSTANCE.setOutputFilter(null);
+            List<TestResult> historicFailures = testState.getHistoricFailures(resultsByClass);
+
+            for (TestRunListener listener : listeners) {
+                listener.runComplete(new TestRunResults(runId, classScanResult, classScanResult == null, start,
+                        System.currentTimeMillis(), toResultsMap(historicFailures, resultsByClass)));
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        } finally {
+            QuarkusConsole.INSTANCE.setOutputFilter(null);
+            Thread.currentThread().setContextClassLoader(old);
+        }
+    }
+
+    public synchronized void abort() {
+        for (TestRunListener listener : listeners) {
+            listener.runAborted();
```

Comment:
Should this be wrapped in try-catch to make sure all the listeners are notified and the code after the loop gets executed?